### PR TITLE
lndclient: add closing txid for waiting close channel response

### PR DESCRIPTION
**Pull request by yyforyongyu**
Bump lnd's version to v0.14.2-beta to include the closing txid from waiting close channel response.

### CODE REVIEW

**Review by carlaKC — COMMENTED** [review]
Should this be on the 14.2 branch? Looks like the relevant [commit](https://github.com/lightningnetwork/lnd/commit/99378b22c3a3b9b7f565f07a2241042c2d6477ec) wasn't included in the release? So I think this should go on a .15 branch, since that's the next release that'll have the field?

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ The current compatibility matrix reads as follows:
 
 | `lndclient` git tag          | `lnd` version in `go.mod` | minimum required `lnd` version | 
 | ---------------------------- | ------------------------- | ------------------------------ |
-| [`v0.14.2-0`](https://github.com/lightninglabs/lndclient/blob/v0.14.2-0) | `lnd @ 4a62b7cf8c69` | `v0.14.0-beta` |
+| [`v0.14.2-1`](https://github.com/lightninglabs/lndclient/blob/v0.14.2-1) | `v0.14.2-beta` | `lnd @ 99378b22c3a3` |
```
Comment (guggero) [inline README.md:29]: nit: can remove/replace the `lnd@...` string and just use `v0.14.2-beta`?

File: lightning_client.go [hunk 2]
```diff
@@ -1845,12 +1848,18 @@ func (s *lightningClient) PendingChannels(ctx context.Context) (*PendingChannels
 			return nil, err
 		}
 
+		hash, err := chainhash.NewHashFromStr(waiting.ClosingTxid)
```
Comment (guggero) [inline lightning_client.go:1851]: Are we sure this is always set? If this was only added in 0.14.2 then we'd need to bump the `minimalCompatibleVersion` to make sure we never run into an error here.

Comment (yyforyongyu) [reply lightning_client.go:1851]: Fixed.

File: README.md (hunk 1 above)
Comment (yyforyongyu) [reply README.md]: Done.

**Comment by yyforyongyu** [comment]
> Should this be on the 14.2 branch? Looks like the relevant [commit](https://github.com/lightningnetwork/lnd/commit/99378b22c3a3b9b7f565f07a2241042c2d6477ec) wasn't included in the release? So I think this should go on a .15 branch, since that's the next release that'll have the field?

It's there actually, but weirdly in a different [commit](https://github.com/lightningnetwork/lnd/commit/981464846a16b92eb02ce19dd09cd125b364814b). Plus the build won't pass if it's not there.

**Review by guggero — APPROVED** [review]

**Review by carlaKC — APPROVED** [review]

**Pull request merged**